pkgs/by-name/cn/cnsprcy/package.nix

```
@@ -5,19 +5,29 @@
   sqlite,
 }:
 
-rustPlatform.buildRustPackage {
+rustPlatform.buildRustPackage (finalAttrs: {
   pname = "cnsprcy";
-  version = "0.2.0";
+  version = "0.3.0";
 
   src = fetchFromSourcehut {
     owner = "~xaos";
     repo = "cnsprcy";
-    rev = "v0.2.0";
-    hash = "sha256-f+DauSU4bT3EljY8/ig7jLnUgyDPEo2NSBQcPN0iKx0=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-OBNl2ZZi1jmJjJvzCwaIN9DSYjGaEo0g4o7/vVOF5aQ=";
   };
 
-  useFetchCargoVendor = true;
-  cargoHash = "sha256-lPTufjKOXMvPy+cP1UyVCTfRXkOmzZqDR6yraIgk+Dg=";
+  cargoLock = {
+    # upstream no longer includes the lockfile
+    lockFile = ./Cargo.lock;
+  };
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
```

We no longer need to specify the `Cargo.lock` file so we can delete it, but we still need to do a few things:

1. Patch the package's lockfile using `cargoPatches` with the registry and checksum of the `cnsprcy` dependency since it doesn't include it by default. This is what the patch file should contain:
```diff
---
 Cargo.lock | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Cargo.lock b/Cargo.lock
index d38ea61..cdb8976 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -294,6 +294,8 @@ dependencies = [
 [[package]]
 name = "cnsprcy"
 version = "0.3.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "e59038606a0721f7fdc920c57ae4b859ee748f13aa2e97dbff52ccd01a5cfe0c"
 dependencies = [
  "bincode",
  "chacha20poly1305",
--
2.47.2
```
2. Change the `sourceRoot` to the directory that contains `Cargo.toml` and `Cargo.lock`

```suggestion
  cargoPatches = [
    # missing upstream
    ./0001-add-cnsprcy-registry-source-and-checksum.patch
  ];

  sourceRoot = "${finalAttrs.src.name}/v${finalAttrs.version}";

  useFetchCargoVendor = true;
  cargoHash = "sha256-TW4hvLBVDIZmCbX4Dw+Gg9km0CSu6MPfpXzx/OiZ4xw=";
```

---

I've been speaking some more with the upstream author (they've followed this PR a little), and they might be able to push another minor version bump to resolve this needing patch.

Of course, happy to merge this now with the patch, and update later alongside upstream.